### PR TITLE
Fix schema inference for *Cluster functions for column names with spaces

### LINKED ISSUE #54615: s3Cluster with schema inference doesn't work if column names contain whitespaces
```
DESCRIBE TABLE s3Cluster(default, 'https://s3.us-east-1.amazonaws.com/icedb-test-tangia-staging/chicago_taxis_1m/_data/**/*.parquet')

Query id: 75a3fcfb-d9ed-49af-bd76-b08169a9f5c3

┌─name───────────────────────┬─type─────────────┬─default_type─┬─default_expression─┬─comment─┬─codec_expression─┬─ttl_expression─┐
│ Trip ID                    │ Nullable(String) │              │                    │         │                  │                │
│ Taxi ID                    │ Nullable(String) │              │                    │         │                  │                │
│ Trip Start Timestamp       │ Nullable(Int64)  │              │                    │         │                  │                │
│ Trip End Timestamp         │ Nullable(String) │              │                    │         │                  │                │
│ Trip Seconds               │ Nullable(String) │              │                    │         │                  │                │
│ Trip Miles                 │ Nullable(String) │              │                    │         │                  │                │
│ Pickup Census Tract        │ Nullable(String) │              │                    │         │                  │                │
│ Dropoff Census Tract       │ Nullable(String) │              │                    │         │                  │                │
│ Pickup Community Area      │ Nullable(String) │              │                    │         │                  │                │
│ Dropoff Community Area     │ Nullable(String) │              │                    │         │                  │                │
│ Fare                       │ Nullable(String) │              │                    │         │                  │                │
│ Tips                       │ Nullable(String) │              │                    │         │                  │                │
│ Tolls                      │ Nullable(String) │              │                    │         │                  │                │
│ Extras                     │ Nullable(String) │              │                    │         │                  │                │
│ Trip Total                 │ Nullable(String) │              │                    │         │                  │                │
│ Payment Type               │ Nullable(String) │              │                    │         │                  │                │
│ Company                    │ Nullable(String) │              │                    │         │                  │                │
│ Pickup Centroid Latitude   │ Nullable(String) │              │                    │         │                  │                │
│ Pickup Centroid Longitude  │ Nullable(String) │              │                    │         │                  │                │
│ Pickup Centroid Location   │ Nullable(String) │              │                    │         │                  │                │
│ Dropoff Centroid Latitude  │ Nullable(String) │              │                    │         │                  │                │
│ Dropoff Centroid Longitude │ Nullable(String) │              │                    │         │                  │                │
│ Dropoff Centroid  Location │ Nullable(String) │              │                    │         │                  │                │
│ _row_id                    │ Nullable(String) │              │                    │         │                  │                │
│ d                          │ Nullable(String) │              │                    │         │                  │                │
└────────────────────────────┴──────────────────┴──────────────┴────────────────────┴─────────┴──────────────────┴────────────────┘

25 rows in set. Elapsed: 0.111 sec. 

clickhouse-cloud :) SELECT count() FROM s3Cluster(default, 'https://s3.us-east-1.amazonaws.com/icedb-test-tangia-staging/chicago_taxis_1m/_data/**/*.parquet')

Received exception from server (version 23.8.1):
Code: 62. DB::Exception: Received from p12uiq1ogd.us-east-2.aws.clickhouse-staging.com:9440. DB::Exception: Received from c-coal-gt-61-server-384r9xm-0.c-coal-gt-61-server-headless.ns-coal-gt-61.svc.cluster.local:9000. DB::Exception: Syntax error (columns declaration list): failed at position 9 ('Nullable'): Nullable(String), Taxi ID Nullable(String), Trip Start Timestamp Nullable(Int64), Trip End Timestamp Nullable(String), Trip Seconds Nullable(String), Trip Miles. Expected one of: COLLATE, NOT, NULL, DEFAULT, MATERIALIZED, ALIAS, EPHEMERAL, AUTO_INCREMENT, COMMENT, CODEC, TTL, PRIMARY KEY, token, Comma, end of query. (SYNTAX_ERROR)
```

### PR DESCRIPTION
**Pull request by Avogar**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix schema inference for *Cluster functions for column names with spaces. Closes https://github.com/ClickHouse/ClickHouse/issues/54615

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit 7e6c41fd153ca9e6a682cd765ae5f91d50eba4ea with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/54635/7e6c41fd153ca9e6a682cd765ae5f91d50eba4ea/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>✅ success</td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>⏳ pending</td></tr>
<tbody>
</table>

File: src/Storages/StorageDictionary.cpp
Comment (antonio2368) [inline src/Storages/StorageDictionary.cpp:1]: Got confused at first why are you modifying `StorageDictionary`

**Review by antonio2368 — APPROVED** [review]

File: src/Storages/StorageDictionary.cpp
Comment (Avogar) [reply src/Storages/StorageDictionary.cpp:1]: Yeah, I think we can actually move this function to a separate file, so it won't be confusing

File: src/Storages/StorageDictionary.cpp [hunk 1]
```diff
@@ -89,7 +89,7 @@ String StorageDictionary::generateNamesAndTypesDescription(const NamesAndTypesLi
     {
         if (!std::exchange(first, false))
             ss << ", ";
-        ss << name_and_type.name << ' ' << name_and_type.type->getName();
+        ss << '`' << name_and_type.name << "` " << name_and_type.type->getName();
```
Comment (alexey-milovidov) [inline src/Storages/StorageDictionary.cpp:92]: This is wrong - no escaping.

See `backQuote`

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]
.

File: src/Storages/StorageDictionary.cpp (hunk 1 above)
Comment (Avogar) [reply src/Storages/StorageDictionary.cpp]: Fixed

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**